omnibus/config/projects/agent.rb

```diff
@@ -105,6 +105,14 @@
   license 'Apache License Version 2.0'
   section 'utils'
   priority 'extra'
+  if ENV.has_key?("KUBERNETES_CPU_REQUEST")
+    compression_threads ENV["KUBERNETES_CPU_REQUEST").to_i
+  end
+  if ENV.has_key?("DEPLOY_AGENT") && ENV["DEPLOY_AGENT"] == "true"
```

Comment:
I believe it should be possible to compute the `compression_level` variable only once at the top of the file and then reuse it throughout the recipe to prevent having the same condition twice.

Reply:
could we even compute it on a common module to prevent duplication between files?

Reply:
done for the computation in the beginning of the file, but there might be a lot of common pieces we could split out indeed